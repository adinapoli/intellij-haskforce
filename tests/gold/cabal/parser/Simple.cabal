-- Initial fooua.cabal generated by cabal init.  For further documentation,
--  see http://haskell.org/cabal/users-guide/

name:                fooua
version:             0.1.0.0a
synopsis:            This is a thing made just fooua!
description:         A description should be long and span multiple lines.  If
                     it doesn't, it couldn't possibly be very descriptive, eh?
                         But then again, the fact that each line is prefixed with
                     whitespace is fun.
license:             Apache-2.0
license-file:        LICENSE
author:              Cary M. Robbins
maintainer:          carymrobbins@gmail.com
-- copyright:           
category:            Concurrency
build-type:          Simple
-- extra-source-files:  
cabal-version:       >=1.10

executable fooua
  main-is:             Main.hs
  -- other-modules:       
  -- other-extensions:

  extensions: TemplateHaskell
              QuasiQuotes

  build-depends:       base >=4 && <5
  -- hs-source-dirs:
  default-language:    Haskell2010
