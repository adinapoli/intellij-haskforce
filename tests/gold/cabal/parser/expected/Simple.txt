Cabal File
  PsiElement(COMMENT)('-- Initial fooua.cabal generated by cabal init.  For further documentation,')
  PsiWhiteSpace('\n')
  PsiElement(COMMENT)('--  see http://haskell.org/cabal/users-guide/')
  PsiWhiteSpace('\n')
  PsiWhiteSpace('\n')
  PACKAGE_NAME
    PsiElement(KEY)('name')
    PsiElement(:)(':')
    FREEFORM
      PsiWhiteSpace('                ')
      PsiElement(VALUE)('fooua')
      PsiWhiteSpace('\n')
  PACKAGE_VERSION
    PsiElement(KEY)('version')
    PsiElement(:)(':')
    NUMBERS
      PsiWhiteSpace('             ')
      PsiErrorElement:Expected numbers
        PsiElement(NUMBER)('0')
        PsiElement(.)('.')
        PsiElement(NUMBER)('1')
        PsiElement(.)('.')
        PsiElement(NUMBER)('0')
        PsiElement(.)('.')
        PsiElement(VALUE_WORD)('0a')
  PsiWhiteSpace('\n')
  CUSTOM_FIELD
    PsiElement(KEY)('synopsis')
    PsiElement(:)(':')
    FREEFORM
      PsiWhiteSpace('            ')
      PsiElement(VALUE)('This is a thing made just fooua!')
      PsiWhiteSpace('\n')
  CUSTOM_FIELD
    PsiElement(KEY)('description')
    PsiElement(:)(':')
    FREEFORM
      PsiWhiteSpace('         ')
      PsiElement(VALUE)('A description should be long and span multiple lines.  If')
      PsiWhiteSpace('\n')
      PsiWhiteSpace('                     ')
      PsiElement(VALUE)('it doesn't, it couldn't possibly be very descriptive, eh?')
      PsiWhiteSpace('\n')
      PsiWhiteSpace('                         ')
      PsiElement(VALUE)('But then again, the fact that each line is prefixed with')
      PsiWhiteSpace('\n')
      PsiWhiteSpace('                     ')
      PsiElement(VALUE)('whitespace is fun.')
      PsiWhiteSpace('\n')
  CUSTOM_FIELD
    PsiElement(KEY)('license')
    PsiElement(:)(':')
    FREEFORM
      PsiWhiteSpace('             ')
      PsiElement(VALUE)('Apache-2.0')
      PsiWhiteSpace('\n')
  CUSTOM_FIELD
    PsiElement(KEY)('license-file')
    PsiElement(:)(':')
    FREEFORM
      PsiWhiteSpace('        ')
      PsiElement(VALUE)('LICENSE')
      PsiWhiteSpace('\n')
  CUSTOM_FIELD
    PsiElement(KEY)('author')
    PsiElement(:)(':')
    FREEFORM
      PsiWhiteSpace('              ')
      PsiElement(VALUE)('Cary M. Robbins')
      PsiWhiteSpace('\n')
  CUSTOM_FIELD
    PsiElement(KEY)('maintainer')
    PsiElement(:)(':')
    FREEFORM
      PsiWhiteSpace('          ')
      PsiElement(VALUE)('carymrobbins@gmail.com')
      PsiWhiteSpace('\n')
  PsiElement(COMMENT)('-- copyright:           ')
  PsiWhiteSpace('\n')
  CUSTOM_FIELD
    PsiElement(KEY)('category')
    PsiElement(:)(':')
    FREEFORM
      PsiWhiteSpace('            ')
      PsiElement(VALUE)('Concurrency')
      PsiWhiteSpace('\n')
  CUSTOM_FIELD
    PsiElement(KEY)('build-type')
    PsiElement(:)(':')
    FREEFORM
      PsiWhiteSpace('          ')
      PsiElement(VALUE)('Simple')
      PsiWhiteSpace('\n')
  PsiElement(COMMENT)('-- extra-source-files:  ')
  PsiWhiteSpace('\n')
  CUSTOM_FIELD
    PsiElement(KEY)('cabal-version')
    PsiElement(:)(':')
    FREEFORM
      PsiWhiteSpace('       ')
      PsiElement(VALUE)('>=1.10')
      PsiWhiteSpace('\n')
  PsiWhiteSpace('\n')
  COMPONENT
    COMPONENT_TYPE
      PsiElement(COMPONENT_TYPE)('executable')
    PsiWhiteSpace(' ')
    COMPONENT_NAME
      PsiElement(COMPONENT_NAME)('fooua')
    PsiWhiteSpace('\n')
    COMPONENT_BODY
      PsiElement(INDENT)('  ')
      CUSTOM_FIELD
        PsiElement(KEY)('main-is')
        PsiElement(:)(':')
        FREEFORM
          PsiWhiteSpace('             ')
          PsiElement(VALUE)('Main.hs')
          PsiWhiteSpace('\n')
      PsiElement(COMMENT)('  -- other-modules:       ')
      PsiWhiteSpace('\n')
      PsiElement(COMMENT)('  -- other-extensions:')
      PsiWhiteSpace('\n')
      PsiWhiteSpace('\n')
      PsiElement(INDENT)('  ')
      CUSTOM_FIELD
        PsiElement(KEY)('extensions')
        PsiElement(:)(':')
        FREEFORM
          PsiWhiteSpace(' ')
          PsiElement(VALUE)('TemplateHaskell')
          PsiWhiteSpace('\n')
          PsiWhiteSpace('              ')
          PsiElement(VALUE)('QuasiQuotes')
          PsiWhiteSpace('\n')
      PsiWhiteSpace('\n')
      PsiElement(INDENT)('  ')
      CUSTOM_FIELD
        PsiElement(KEY)('build-depends')
        PsiElement(:)(':')
        FREEFORM
          PsiWhiteSpace('       ')
          PsiElement(VALUE)('base >=4 && <5')
          PsiWhiteSpace('\n')
      PsiElement(COMMENT)('  -- hs-source-dirs:')
      PsiWhiteSpace('\n')
      PsiElement(INDENT)('  ')
      CUSTOM_FIELD
        PsiElement(KEY)('default-language')
        PsiElement(:)(':')
        FREEFORM
          PsiWhiteSpace('    ')
          PsiElement(VALUE)('Haskell2010')